tests/BoltOn.Tests/Data/CosmosDb/RepositoryTests.cs

```diff
@@ -0,0 +1,219 @@
+using System;
+using Microsoft.Extensions.DependencyInjection;
+using Xunit;
+using BoltOn.Bootstrapping;
+using System.Threading.Tasks;
+using System.Linq;
+using BoltOn.Tests.Other;
+using BoltOn.Data.CosmosDb;
+
+namespace BoltOn.Tests.Data.CosmosDb
+{
+    // Collection is used to prevent running tests in parallel i.e., tests in the same collection
+    // will not be executed in parallel
+    [Collection("IntegrationTests")]
+    public class RepositoryTests : IDisposable
+    {
+        private readonly IGradeRepository _sut;
+
+        public RepositoryTests()
+        {
+            var serviceCollection = new ServiceCollection();
+            serviceCollection
+                .BoltOn(options =>
+                {
+                    options
+                        .BoltOnCosmosDbModule();
+                });
+            var serviceProvider = serviceCollection.BuildServiceProvider();
+            serviceProvider.TightenBolts();
+      
+            _sut = serviceProvider.GetService<IGradeRepository>();
+        }
+
+        [Fact, Trait("Category", "Integration")]
+        public void GetById_WhenRecordExists_ReturnsRecord()
+        {
+            // arrange
+            var id = Guid.NewGuid().ToString();
+            var grade = new Grade { Id = id, StudentId = 1, CourseName = "physics", Score = "A-", Year = 2017 };
+            _sut.Add(grade);
```

Comment:
why don't you add and delete in a PostRegistrationTask, doing it within GetById test confuses whether the test is for GetById or Add or Delete